### ISSUE TITLE
Different file extensions for temporary and registered wikis

### ISSUE THREAD

**Issue by thezeroalpha** [vimwiki/vimwiki#869]
It would be nice to allow the user to register wikis in `g:vimwiki_list` with specific extensions, but then only allow creation of temporary wikis with different extensions.

For example, I like my 'main' wiki to have the extension `.txt`, because this makes it automatically and easily readable on any device, so I have it defined as such in `g:vimwiki_list`:

```
let g:vimwiki_list = { 'path': '/home/zeroalpha/Dropbox/vimwiki/', 'ext': '.txt', 'syntax': 'default'}
```

However, this has the effect of interpreting all `.txt` files as vimwiki files. As far as I know from past discussions, the only option is to turn off `g:vimwiki_global_ext`; however, I want to also be able to edit arbitrary `.wiki` files as vimwiki files.

**TLDR**: I want files with the `.txt` extension in the path in `g:vimwiki_list` to be treated as vimwiki files. I don't want any other `.txt` files to be treated as vimwiki files, but I do want this for arbitrary `.wiki` files.

I implemented [a hacky solution in my fork of vimwiki](https://github.com/thezeroalpha/vimwiki/blob/e66d8016220ff67da8e036906f5084c76c491a76/plugin/vimwiki.vim#L75). The main idea is that I interpret `g:vimwiki_ext2syntax` also as a list of extensions that are valid for temporary wikis. In my case, I have `let g:vimwiki_ext2syntax = {'.wiki': 'default'}` in my .vimrc.

Could a feature like this perhaps be useful for more users? I'm considering creating a PR, but I'm not sure whether other users would benefit from this.
